plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/storage/KVMStoragePoolManager.java

```diff
@@ -113,7 +113,8 @@ public KVMStoragePoolManager(StorageLayer storagelayer, KVMHAMonitor monitor) {
                     s_logger.warn(String.format("Duplicate StorageAdaptor type %s, not loading %s", info.storagePoolType().toString(), storageAdaptor.getName()));
                 } else {
                     try {
-                        this._storageMapper.put(info.storagePoolType().toString(), storageAdaptor.newInstance());
+                        s_logger.info(String.format("adding storage adaptor for %s", storageAdaptor.getName()));
+                        this._storageMapper.put(info.storagePoolType().toString(), storageAdaptor.getDeclaredConstructor().newInstance());
```

Comment:
@DaanHoogland pardon my ignorance, what does getDeclaredConstructor do?

Reply:
`getDeclaredConstructor().newInstance()` does the same as `newInstance()` but in a non-deprecated way. I haven´t figured it out yet, but I don´t think this will solve the issue.